benchmark/syclblas/blas1/axpy.cpp

```diff
@@ -96,26 +111,35 @@ void run(benchmark::State& state, blas::SB_Handle* sb_handle_ptr, index_t size,
   }
 
   blas_benchmark::utils::calc_avg_counters(state);
+
+  blas::helper::deallocate<mem_alloc>(inx, q);
+  blas::helper::deallocate<mem_alloc>(iny, q);
 }
 
 template <typename scalar_t>
-void register_benchmark(blas_benchmark::Args& args, blas::SB_Handle* sb_handle_ptr,
-                        bool* success) {
+void register_benchmark(blas_benchmark::Args& args,
+                        blas::SB_Handle* sb_handle_ptr, bool* success) {
   auto axpy_params = blas_benchmark::utils::get_blas1_params(args);
 
-  for (auto size : axpy_params) {
-    auto BM_lambda = [&](benchmark::State& st, blas::SB_Handle* sb_handle_ptr,
-                         index_t size, bool* success) {
-      run<scalar_t>(st, sb_handle_ptr, size, success);
-    };
-    benchmark::RegisterBenchmark(get_name<scalar_t>(size).c_str(), BM_lambda,
-                                 sb_handle_ptr, size, success);
+#define REGISTER_MEM_TYPE_BENCHMARKS(MEMORY, NAME)                             \
```

Comment:
Remove macro, create a function that accepts `AllocType` as template parameter and then call this function twice in the loop.